src/auth/Crypto.cc

```diff
@@ -233,30 +316,18 @@ class CryptoAESKeyHandler : public CryptoKeyHandler {
     ceph::bufferptr out_tmp{static_cast<unsigned>(
       AES_BLOCK_LEN + p2align(in.length(), AES_BLOCK_LEN))};
 
-    // let's pad the data
-    std::uint8_t pad_len = out_tmp.length() - in.length();
-    ceph::bufferptr pad_buf{pad_len};
-    memset(pad_buf.c_str(), pad_len, pad_len);
-
     // form contiguous buffer for block cipher. The ctor copies shallowly.
+    // TODO: encrypt each bufferptr separately if possible?
     ceph::bufferlist incopy(in);
-    incopy.append(std::move(pad_buf));
-    const auto in_buf = reinterpret_cast<unsigned char*>(incopy.c_str());
-
-    // reinitialize IV each time. It might be unnecessary depending on
-    // actual implementation but at the interface layer we are obliged
-    // to deliver IV as non-const.
-    static_assert(strlen_ct(CEPH_AES_IV) == AES_BLOCK_LEN);
-    unsigned char iv[AES_BLOCK_LEN];
-    memcpy(iv, CEPH_AES_IV, AES_BLOCK_LEN);
-
-    // we aren't using EVP because of performance concerns. Profiling
-    // shows the cost is quite high. Endianness might be an issue.
```

Comment:
This might be not valid anymore as Cephx's signature size has grown. More investigation and testing needed. WIP.

BTW: I can recall that ISAL Crypto was also considered during the transition. It doesn't have required certifications but maybe having it as an optional implementation would be possible. The technical pros are: it's already in the tree, offers very good performance and doesn't need malloc. Just thinking loudly.